src/Storages/NATS/NATSSettings.h

```diff
@@ -17,7 +17,7 @@ class ASTStorage;
     M(UInt64, nats_num_consumers, 1, "The number of consumer channels per table.", 0) \
     M(String, nats_queue_group, "", "Name for queue group of NATS subscribers.", 0) \
     M(Bool, nats_secure, false, "Use SSL connection", 0) \
-    M(UInt64, nats_max_reconnect, 5, "Maximum amount of reconnection attempts.", 0) \
+    M(UInt64, nats_max_reconnect, 5, "Deprecated and has no effect, reconnect is performed permanently with nats_reconnect_wait timeout", 0) \
```

Comment:
If this is not used, then maybe let's remove all references to it from the code to make more clear.

Reply:
I have kept this setting for backward compatibility. If we remove only this parameter, we will not be able to load previously created tables after server update.

I can add ignoring the nats_max_reconnect to the `NATSSettings::loadFromQuery` method:
```C++
void NATSSettings::loadFromQuery(ASTStorage & storage_def)
{
    if (storage_def.settings)
    {
        try
        {
            storage_def.settings->changes.removeSetting("nats_max_reconnect");
            applyChanges(storage_def.settings->changes);
        }
        catch (Exception & e)
        {
            if (e.code() == ErrorCodes::UNKNOWN_SETTING)
                e.addMessage("for storage " + storage_def.engine->name);
            throw;
        }
    }
    ....
}
```
But in this case we won't be able to get an error if we pass nats_max_reconnect in the CREATE TABLE query after updating the server.

Which behavior is better?

Reply:
Definitely leave the setting here for the compatibility reason you mentioned. I phrased my comment poorly, but what I meant is remove every occasion of accessing this setting, e.g.: https://github.com/ClickHouse/ClickHouse/blob/1f1a9c8508696a8f81740f6a49c5ebca79622c74/src/Storages/NATS/StorageNATS.cpp#L90 , and also from the `NATSConfiguration` struct https://github.com/ClickHouse/ClickHouse/blob/1f1a9c8508696a8f81740f6a49c5ebca79622c74/src/Storages/NATS/NATSConnection.h#L19

Because right now the docs mention that it is deprecated, but I still see the usage of it.

Reply:
I am fixed it